styx-api-service/src/test/java/com/spotify/styx/api/WorkflowResourceTest.java

```diff
@@ -203,61 +187,44 @@ public void shouldSucceedWithFullPatchStatePerWorkflow() throws Exception {
     assertJson(response, "next_natural_trigger", equalTo("2016-08-10T07:00:01Z"));
     assertJson(response, "next_natural_offset_trigger", equalTo("2016-08-10T08:00:01Z"));
 
-    assertThat(storage.enabled(WORKFLOW.id()), is(true));
-    assertThat(storage.workflowState(WORKFLOW.id()).nextNaturalTrigger().get().toString(),
+    final WorkflowState workflowState = storage.workflowState(WORKFLOW.id());
+    assertThat(workflowState.enabled().get(), is(true));
+    assertThat(workflowState.nextNaturalTrigger().get().toString(),
                equalTo("2016-08-10T07:00:01Z"));
-    assertThat(storage.workflowState(WORKFLOW.id()).nextNaturalOffsetTrigger().get().toString(),
+    assertThat(workflowState.nextNaturalOffsetTrigger().get().toString(),
                equalTo("2016-08-10T08:00:01Z"));
   }
 
   @Test
   public void shouldSucceedWithEnabledPatchStatePerWorkflow() throws Exception {
     sinceVersion(Api.Version.V3);
 
-    storage.patchState(WORKFLOW.id(), patchDockerImage("preset:image"));
-
     Response<ByteString> response =
         awaitResponse(serviceHelper.request("PATCH", path("/foo/bar/state"),
                                             STATEPAYLOAD_ENABLED));
 
     assertThat(response, hasStatus(withCode(Status.OK)));
     assertThat(response, hasHeader("Content-Type", equalTo("application/json")));
     assertJson(response, "enabled", equalTo(true));
-    assertJson(response, "docker_image", equalTo("preset:image"));
 
     assertThat(storage.enabled(WORKFLOW.id()), is(true));
-    assertThat(storage.getDockerImage(WORKFLOW.id()), is(Optional.of("preset:image")));
   }
 
   @Test
   public void shouldSucceedWhenStatePayloadWithOtherFieldsIsSent() throws Exception {
     sinceVersion(Api.Version.V3);
 
-    storage.patchState(WORKFLOW.id(), patchDockerImage("preset:image"));
-
     Response<ByteString> response =
         awaitResponse(serviceHelper.request("PATCH", path("/foo/bar/state"),
                                             STATEPAYLOAD_OTHER_FIELD));
 
     assertThat(response, hasStatus(withCode(Status.OK)));
-    assertThat(response, hasHeader("Content-Type", equalTo("application/json")));
```

Comment:
Did you mean to remove this assertion?

Reply:
it just didn't seem necessary, there are other checks for that. i can add back if you want.